src/mon/AuthMonitor.cc

```diff
@@ -1927,6 +1902,68 @@ int AuthMonitor::_check_and_encode_caps(const map<string, string>& caps,
   return 0;
 }
 
+/* Pass both, rdata as well as fmtr to enable printing of the key after
+ * update and set create to True to allow authorizing a new entity instead
+ * of updating its caps. */
+int AuthMonitor::_update_or_create_entity(const EntityName& entity,
+  const map<string, string>& caps, MonOpRequestRef op, stringstream& ds,
+  bufferlist* rdata, Formatter* fmtr, bool create_entity)
+{
+  stringstream ss;
+  KeyServerData::Incremental auth_inc;
+  auth_inc.name = entity;
+
+  if (!create_entity &&
+      !mon.key_server.get_auth(auth_inc.name, auth_inc.auth)) {
+    ss << "couldn't find entry " << auth_inc.name;
+    return -ENOENT;
+  }
+
+  map<string, bufferlist> encoded_caps;
+  if (_check_and_encode_caps(caps, encoded_caps, ss) == -EINVAL) {
+    return -EINVAL;
```

Comment:
```suggestion
  if (auto err = _check_and_encode_caps(caps, encoded_caps, ss); err < 0) {
    return err;
```

Reply:
Why create a variable (`err`) when we don't use it anywhere in the function?

Reply:
We actually use it in `return err` for lowering the coupling between the caller and callee. Now the contract is that literally the very sole allowed error in `_check_and_encode_caps()` is `EINVAL`. If another one is added, all callers must be reworked which is errorprone.

Reply:
I had a discussion about this with @rzarzynski. This change has been made